sharding/database/inmemory_test.go

```diff
@@ -0,0 +1,70 @@
+package database
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
```

Comment:
Please add a test that this database properly implements the ShardBackend

Example: https://github.com/ethereum/go-ethereum/blob/master/ethclient/ethclient_test.go